test/kmp_matcher.lean

```diff
@@ -0,0 +1,23 @@
+import Std.Tactic.GuardMsgs
+import Std.Data.String.Basic
+
+/-! Tests for Knuth-Morris-Pratt matching algorithm -/
```

Comment:
Wouldn't hurt for the tests to directly exercise `String.findSubstr?`, given that's the main entrypoint.